src/proto/streams/store.rs

```diff
@@ -82,35 +83,32 @@ impl Store {
         Store {
             slab: slab::Slab::new(),
             ids: IndexMap::new(),
-            counter: 0,
         }
     }
 
     pub fn find_mut(&mut self, id: &StreamId) -> Option<Ptr> {
-        let key = match self.ids.get(id) {
+        let index = match self.ids.get(id) {
             Some(key) => *key,
             None => return None,
         };
 
         Some(Ptr {
             key: Key {
-                index: key.0,
-                store_id: key.1,
+                index,
+                stream_id: *id,
             },
             store: self,
         })
     }
 
     pub fn insert(&mut self, id: StreamId, val: Stream) -> Ptr {
-        let store_id = self.counter;
-        self.counter = self.counter.wrapping_add(1);
-        let key = self.slab.insert((store_id, val));
-        assert!(self.ids.insert(id, (key, store_id)).is_none());
+        let idx = SlabIndex(self.slab.insert(val));
+        assert!(self.ids.insert(id, idx).is_none());
 
         Ptr {
             key: Key {
-                index: key,
-                store_id,
+                index: idx,
```

Comment:
TIOLI: why not just call the local var `index` so this can use local var shorthand?